tutorials/hello/hello_client/src/app/plugins/hello_info/hello_info_plugin.ts

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable prefer-destructuring */
+/* eslint-disable no-loop-func */
+/* eslint-disable @typescript-eslint/prefer-for-of */
+/* eslint-disable for-direction */
+/* eslint-disable no-param-reassign */
+/* eslint-disable @typescript-eslint/no-floating-promises */
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+/* eslint-disable consistent-return */
+/* eslint-disable no-useless-return */
+/* eslint-disable import/namespace */
+/* eslint-disable @typescript-eslint/prefer-readonly */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable dot-notation */
+/* eslint-disable @typescript-eslint/require-await */
+/* eslint-disable no-console */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable spaced-comment */
+/* eslint-disable prefer-const */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable class-methods-use-this */
+/* eslint-disable  @typescript-eslint/no-empty-function */
+import { BasePlugin, db as liskDB, PluginInitContext } from 'lisk-sdk';
+import {
+	getDBInstance,
+	setLastCounter,
+	getLastCounter,
+} from './db';
+import { configSchema } from './schemas';
+import { HelloInfoPluginConfig } from './types';
+import { Endpoint } from './endpoint';
+
+export class HelloInfoPlugin extends BasePlugin<HelloInfoPluginConfig> {
+	public configSchema = configSchema;
+	public endpoint = new Endpoint();
+	public counter = 0;
+	public height = 0;
+	private _addressPluginDB!: liskDB.Database;
+
+	public get nodeModulePath(): string {
+		return __filename;
+	}
+
+	public async init(context: PluginInitContext): Promise<void> {
```

Comment:
maybe in init you can check if lastCounter is not assigned or you get NotFoundError, if not assigned then you can initialize to zero?

Reply:
Wouldn't I have to move my `this._addressPluginDB = await getDBInstance(this.dataPath);` to the `init()` function as well? If I want to check `lastCounter` from DB?

Does the `init()` function work exactly like `load()`?